fvm/evm/emulator/emulator.go

```diff
@@ -400,15 +437,16 @@ func (proc *procedure) run(
 		gasPool,
 	).TransitionDb()
 	if err != nil {
-		// if the error is a fatal error or a non-fatal state error return it
-		// this condition should never happen
-		// given all StateDB errors are withheld for the commit time.
-		if types.IsAFatalError(err) || types.IsAStateError(err) {
-			return &res, err
+		// if the error is a fatal error or a non-fatal state error or a backend err return it
+		// this condition should never happen given all StateDB errors are withheld for the commit time.
+		if types.IsAFatalError(err) || types.IsAStateError(err) || types.IsABackendError(err) {
+			return nil, err
 		}
 		// otherwise is a validation error (pre-check failure)
 		// no state change, wrap the error and return
-		return &res, types.NewEVMValidationError(err)
+		res.ValidationError = err
```

Comment:
should validation error better have a setter method that internally sets the gas cost? I fear someone can set the validation error but forgets to set the gas cost, so something like `res.SetValidationError(err)` could internally handle that.

Reply:
good idea, going to update it.

Reply:
updated